src/api/model/modelApi.ts

```diff
@@ -14,57 +14,71 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
+  async installModels(
+    database: string,
+    engine: string,
+    models: Model[],
+    async = false,
```

Comment:
Are we doing via a param in the other SDKs?

This kinda makes it hard for Typescript.

![image](https://user-images.githubusercontent.com/6468871/190690241-14c642f7-7971-4361-af0b-ca6b70337326.png)

I can't access `results` because the return type get inferred from `execAsync`. Can we do it in a separate function? We do have `exec` and `execAsync` after all.

If no, we can annotate the return type better. Like:

```
async installModels(
    database: string,
    engine: string,
    models: Model[],
    async = false,
  ): Promise<TransactionAsyncResult | NEED_NEW_TYPE_HERE> {
```

where `NEED_NEW_TYPE_HERE` is what it has now. `{transaction: ...}`

Reply:
I think it's better adding new functions :pray: